slasher/db/iface/interface.go

```diff
@@ -0,0 +1,59 @@
+package iface
+
+import (
+	"io"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	slashpb "github.com/prysmaticlabs/prysm/proto/slashing"
+	"github.com/prysmaticlabs/prysm/slasher/db/types"
+)
+
+// SlasherDB defines the necessary methods for a Prysm slasher DB.
+type SlasherDB interface {
```

Comment:
Would be good to separate this to read only and read + write

Reply:
+1 , define different interfaces for different access patterns